librz/include/rz_lib.h

```diff
@@ -31,95 +29,111 @@ RZ_LIB_VERSION_HEADER(rz_lib);
 #define RZ_LIB_EXT "so"
 #endif
 
-/* store list of loaded plugins */
+/**
+ * \brief Identify the Plugin type
+ */
+typedef enum {
+	RZ_LIB_TYPE_IO, ///< io layer
+	RZ_LIB_TYPE_DBG, ///< debugger
+	RZ_LIB_TYPE_LANG, ///< language
+	RZ_LIB_TYPE_ASM, ///< assembler
+	RZ_LIB_TYPE_ANALYSIS, ///< analysis
+	RZ_LIB_TYPE_PARSE, ///< parsers
+	RZ_LIB_TYPE_BIN, ///< bin headers
+	RZ_LIB_TYPE_BIN_XTR, ///< bin extractors
+	RZ_LIB_TYPE_BIN_LDR, ///< DEPRECATED
+	RZ_LIB_TYPE_BP, ///< breakpoint
+	RZ_LIB_TYPE_SYSCALL, ///< DEPRECATED
+	RZ_LIB_TYPE_FASTCALL, ///< DEPRECATED
+	RZ_LIB_TYPE_CRYPTO, ///< cryptography
+	RZ_LIB_TYPE_HASH, ///< hashes / message digests
+	RZ_LIB_TYPE_CORE, ///< RzCore commands
+	RZ_LIB_TYPE_EGG, ///< rz_egg plugin
+	RZ_LIB_TYPE_DEMANGLER, ///< demanglers
+	RZ_LIB_TYPE_UNKNOWN
+} RzLibType;
+
+/**
+ * \brief Represent a single Plugin
+ */
 typedef struct rz_lib_plugin_t {
-	int type;
-	char *file;
-	void *data; /* user pointer */
-	struct rz_lib_handler_t *handler;
-	void *dl_handler; // DL HANDLER
-	char *author;
-	char *version;
+	RzLibType type; ///< Type of the plugin
+	char *file; ///< File path
+	void *data; ///< User pointer
+	struct rz_lib_handler_t *handler; ///< Handler that handles this plugin
+	char *author; ///< Author of the plugin
+	char *version; ///< Version of the plugin
 	void (*free)(void *data);
 } RzLibPlugin;
 
-/* store list of initialized plugin handlers */
+typedef bool (*RzLibCallback)(RzLibPlugin *, void *, void *);
+
+/**
+ * \brief Identify how a type of plugins should be handled.
+ */
 typedef struct rz_lib_handler_t {
-	int type;
-	char desc[128]; // TODO: use char *
-	void *user; /* user pointer */
-	int (*constructor)(RzLibPlugin *, void *user, void *data);
-	int (*destructor)(RzLibPlugin *, void *user, void *data);
+	RzLibType type; ///< Plugin type this handler handles
+	char desc[128]; ///< Description of the handler
+	void *user; ///< User pointer
```

Comment:
As said before, this is never going to be populated

Reply:
This is populated by rz_lib_add_handler and it is passed as the second argument of ->constructor. What is the problem there?